### PR TITLE
Wicked Sales JS: Add Item To Cart Back End | Keith Tachibana

### PR DESCRIPTION
**Pull request by Keith-Tachibana**
image

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -35,10 +35,8 @@ app.get('/api/products', (req, res, next) => {
 
 app.get('/api/products/:productId', (req, res, next) => {
   const { productId } = req.params;
-  if (!parseInt(productId, 10)) {
-    return res.status(400).json({
-      error: 'The productId must be a positive integer.'
-    });
+  if ((!parseInt(productId, 10)) || (parseInt(productId, 10) < 1)) {
+    next(new ClientError('The productId must be a positive integer', 400));
```
Comment (thebearingedge) [inline server/index.js]: Be sure to `return` here so that your middleware doesn't continue to execute.

File: server/index.js [hunk 2]
```diff
@@ -58,6 +56,102 @@ app.get('/api/products/:productId', (req, res, next) => {
...
+      .catch(err => next(err));
+  } else {
+    res.status(200).json([]);
+  }
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const { productId } = req.body;
+  if ((!parseInt(productId, 10)) || (parseInt(productId, 10) < 1)) {
+    next(new ClientError('The productId must be a positive integer', 400));
```
Comment (thebearingedge) [inline server/index.js]: Be sure to `return` here.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: server/index.js (hunk 1 above)
Comment (Keith-Tachibana) [reply server/index.js]: Fixed

File: server/index.js (hunk 2 above)
Comment (Keith-Tachibana) [reply server/index.js]: Fixed

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**